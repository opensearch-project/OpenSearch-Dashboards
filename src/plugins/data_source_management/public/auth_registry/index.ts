/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export {
  IAuthenticationMethodRegistry,
  AuthenticationMethod,
  AuthenticationMethodRegistry,
} from './authentication_methods_registry';
