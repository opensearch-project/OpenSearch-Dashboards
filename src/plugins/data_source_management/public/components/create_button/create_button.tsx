/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { History } from 'history';

import { EuiButton } from '@elastic/eui';
import { FormattedMessage } from '@osd/i18n/react';

interface Props {
  history: History;
  isEmptyState?: boolean;
  dataTestSubj: string;
}

export const CreateButton = ({ history, isEmptyState, dataTestSubj }: Props) => {
  return (
    <EuiButton
      data-test-subj={dataTestSubj}
      fill={isEmptyState ? false : true}
      onClick={() => history.push('/create')}
    >
      <FormattedMessage
        id="dataSourcesManagement.dataSourceListing.createButton"
        defaultMessage="Create data source connection"
      />
    </EuiButton>
  );
};
