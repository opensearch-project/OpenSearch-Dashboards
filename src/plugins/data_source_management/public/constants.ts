/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DirectQueryDatasourceType } from './types';

export const QUERY_RESTRICTED = 'query-restricted';
export const QUERY_ALL = 'query-all';

export const DatasourceTypeToDisplayName: { [key in DirectQueryDatasourceType]: string } = {
  PROMETHEUS: 'Prometheus',
  S3GLUE: 'Amazon S3',
};

export const PROMETHEUS_URL = 'Prometheus';
export const AMAZON_S3_URL = 'AmazonS3AWSGlue';
export const OPENSEARCH_URL = 'OpenSearch';

export const UrlToDatasourceType: { [key: string]: DirectQueryDatasourceType } = {
  [PROMETHEUS_URL]: 'PROMETHEUS',
  [AMAZON_S3_URL]: 'S3GLUE',
};
