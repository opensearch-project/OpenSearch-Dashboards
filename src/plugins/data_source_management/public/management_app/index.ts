/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { mountManagementSection } from './mount_management_section';
