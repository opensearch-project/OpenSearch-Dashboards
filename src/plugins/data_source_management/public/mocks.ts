/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { throwError } from 'rxjs';
import { SavedObjectsClientContract } from 'opensearch-dashboards/public';
import { AuthType } from './types';
import { coreMock } from '../../../core/public/mocks';

/* Mock Types */

export const docLinks = {
  links: {
    noDocumentation: {
      indexPatterns: {
        introduction: '',
      },
      scriptedFields: {},
    },
  },
};

const createDataSourceManagementContext = () => {
  const {
    chrome,
    application,
    savedObjects,
    uiSettings,
    notifications,
    overlays,
  } = coreMock.createStart();
  const { http } = coreMock.createSetup();

  return {
    chrome,
    application,
    savedObjects,
    uiSettings,
    notifications,
    overlays,
    http,
    docLinks,
    setBreadcrumbs: () => {},
  };
};

export const mockManagementPlugin = {
  createDataSourceManagementContext,
  docLinks,
};

/* Mock data responses - JSON*/
export const getDataSourcesResponse = {
  savedObjects: [
    {
      id: 'test',
      type: 'data-source',
      description: 'test datasource',
      title: 'test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
    {
      id: 'test2',
      type: 'data-source',
      description: 'test datasource2',
      title: 'test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
    {
      id: 'alpha-test',
      type: 'data-source',
      description: 'alpha test datasource',
      title: 'alpha-test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
    {
      id: 'beta-test',
      type: 'data-source',
      description: 'beta test datasource',
      title: 'beta-test',
      get(field: string) {
        const me: any = this || {};
        return me[field];
      },
    },
  ],
};

export const existingDatasourceNamesList = [
  'test123',
  'testTest20',
  'TeSt',
  'duplicateTest',
  'dup20',
];

export const getMappedDataSources = [
  {
    id: 'test',
    description: 'test datasource',
    title: 'test',
    sort: 'test',
  },
  {
    id: 'test2',
    description: 'test datasource2',
    title: 'test',
    sort: 'test',
  },
  {
    id: 'alpha-test',
    description: 'alpha test datasource',
    title: 'alpha-test',
    sort: 'alpha-test',
  },
  {
    id: 'beta-test',
    description: 'beta test datasource',
    title: 'beta-test',
    sort: 'beta-test',
  },
];

export const mockDataSourceAttributesWithAuth = {
  id: 'test',
  title: 'create-test-ds',
  description: 'jest testing',
  endpoint: 'https://test.com',
  auth: {
    type: AuthType.UsernamePasswordType,
    credentials: {
      username: 'test123',
      password: 'test123',
    },
  },
};

export const mockDataSourceAttributesWithNoAuth = {
  id: 'test123',
  title: 'create-test-ds123',
  description: 'jest testing',
  endpoint: 'https://test.com',
  auth: {
    type: AuthType.NoAuth,
    credentials: undefined,
  },
};
export const getDataSourceByIdWithCredential = {
  attributes: {
    id: 'alpha-test',
    title: 'alpha-test',
    endpoint: 'https://test.com',
    description: 'alpha test data source',
    auth: {
      type: AuthType.UsernamePasswordType,
      credentials: {
        username: 'test123',
      },
    },
  },
};

export const getDataSourceByIdWithoutCredential = {
  attributes: {
    ...getDataSourceByIdWithCredential.attributes,
    auth: {
      type: AuthType.NoAuth,
      credentials: undefined,
    },
  },
  references: [],
};

export const mockResponseForSavedObjectsCalls = (
  savedObjectsClient: SavedObjectsClientContract,
  savedObjectsMethodName: 'get' | 'find' | 'create' | 'delete' | 'update',
  response: any
) => {
  (savedObjectsClient[savedObjectsMethodName] as jest.Mock).mockResolvedValue(response);
};

export const mockErrorResponseForSavedObjectsCalls = (
  savedObjectsClient: SavedObjectsClientContract,
  savedObjectsMethodName: 'get' | 'find' | 'create' | 'delete' | 'update'
) => {
  (savedObjectsClient[savedObjectsMethodName] as jest.Mock).mockRejectedValue(
    throwError(new Error('Error while fetching data sources'))
  );
};
