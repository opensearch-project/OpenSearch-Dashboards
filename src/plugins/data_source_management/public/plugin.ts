/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CoreSetup, CoreStart, Plugin } from '../../../core/public';

import { PLUGIN_NAME } from '../common';

import { ManagementSetup } from '../../management/public';
import { IndexPatternManagementSetup } from '../../index_pattern_management/public';
import { DataSourceColumn } from './components/data_source_column/data_source_column';

export interface DataSourceManagementSetupDependencies {
  management: ManagementSetup;
  indexPatternManagement: IndexPatternManagementSetup;
}

const DSM_APP_ID = 'dataSources';

export class DataSourceManagementPlugin
  implements Plugin<void, void, DataSourceManagementSetupDependencies> {
  public setup(
    core: CoreSetup,
    { management, indexPatternManagement }: DataSourceManagementSetupDependencies
  ) {
    const opensearchDashboardsSection = management.sections.section.opensearchDashboards;

    if (!opensearchDashboardsSection) {
      throw new Error('`opensearchDashboards` management section not found.');
    }

    const savedObjectPromise = core
      .getStartServices()
      .then(([coreStart]) => coreStart.savedObjects);
    const httpPromise = core.getStartServices().then(([coreStart]) => coreStart.http);
    const column = new DataSourceColumn(savedObjectPromise, httpPromise);
    indexPatternManagement.columns.register(column);

    opensearchDashboardsSection.registerApp({
      id: DSM_APP_ID,
      title: PLUGIN_NAME,
      order: 1,
      mount: async (params) => {
        const { mountManagementSection } = await import('./management_app');

        return mountManagementSection(core.getStartServices, params);
      },
    });
  }

  public start(core: CoreStart) {}

  public stop() {}
}
