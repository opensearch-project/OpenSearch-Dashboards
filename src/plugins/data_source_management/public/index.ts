/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DataSourceManagementPlugin } from './plugin';

// This exports static code and TypeScript types,
// as well as, OpenSearch Dashboards Platform `plugin()` initializer.
export function plugin() {
  return new DataSourceManagementPlugin();
}
export { DataSourceManagementPluginStart } from './types';
