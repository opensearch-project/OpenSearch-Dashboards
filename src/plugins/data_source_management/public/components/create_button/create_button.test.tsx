/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CreateButton } from './create_button';
import { scopedHistoryMock } from '../../../../../core/public/mocks';
import { ScopedHistory } from 'opensearch-dashboards/public';

const createButtonIdentifier = `[data-test-subj="createDataSourceButton"]`;

describe('CreateButton', () => {
  const history = (scopedHistoryMock.create() as unknown) as ScopedHistory;
  const dataTestSubj = 'createDataSourceButton';
  let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    component = shallow(<CreateButton history={history} dataTestSubj={dataTestSubj} />);
  });

  it('should render normally', () => {
    expect(component).toMatchSnapshot();
  });

  it('should click event normally', () => {
    component.find(createButtonIdentifier).first().simulate('click');

    expect(history.push).toBeCalledWith('/create');
  });
});
