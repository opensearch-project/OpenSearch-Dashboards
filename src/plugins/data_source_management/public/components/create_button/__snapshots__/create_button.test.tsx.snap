// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateButton should render normally 1`] = `
<EuiButton
  data-test-subj="createDataSourceButton"
  fill={true}
  onClick={[Function]}
>
  <FormattedMessage
    defaultMessage="Create data source connection"
    id="dataSourcesManagement.dataSourceListing.createButton"
    values={Object {}}
  />
</EuiButton>
`;
